benchadapt/python/benchadapt/result.py

```diff
@@ -171,10 +173,20 @@ def _cluster_info_property(self, value: Dict[str, Any]) -> None:
             self.machine_info = None
         self._cluster_info_cache = value
 
-    def to_publishable_dict(self):
-        """Returns a dict suitable for sending to conbench"""
+    def to_publishable_dict(self) -> Dict:
+        """
+        Return a dictionary representing the benchmark result.
+
+        After JSON-serialization, that dictionary is expected to validate
+        against the JSON schema that the Conbench API expects on the endpoint
+        for benchmark result submission.
+        """
+
         res_dict = asdict(self)
 
+        # We should discuss why we don't exit with an error here (publish this
+        # although it's not publishable? who consumes the warning? should the
+        # warning be re-worded to be more user-friendly?)
```

Comment:
I think this is really important. If I imagine to build client tooling and seeing a warning like "Result not publishable!" but  then everything "still works" is kind of confusing.

Reply:
Hm, I thought we had a discussion about this somewhere on GitHub but can't find it.

Reply:
Thanks for trying to find it!